Community/Tdarr_Plugin_goof1_URL_Plex_Refresh.js

```diff
@@ -0,0 +1,221 @@
+
+              If your TDarr path is:
+              /media/local/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+              
+              And the Plex path to the file is:
+              /data/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+
+              then part you would enter here is:
+
+               \\nExample:\\n
+               /media/local/`,
+    },
+  ],
+});
+
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+  inputs = loadDefaultValues(inputs, details);
+
+  let response = {
+    file,
+    removeFromDB: false,
+    updateDB: false,
+    infoLog: '',
+  };
+
+  const http = require('http')
+  const https = require('https')
+
+  const type = inputs.Url_Protocol
+  const url = inputs.Plex_Url
+  const port = inputs.Plex_Port
+  const token = inputs.Plex_Token
+  const key = inputs.Library_Key
+  const plexPath = inputs.Plex_Path
+  const tdarrPath = inputs.TDarr_Path
+
+  if (!type || !url || !token || !key) {
+    response.infoLog = `Url_Protocol, Plex_Url, Plex_Token, and Library_Key are all required`
+    response.processFile = false
+    return response
+  }
+
+
+  let filePath = file.file
+  filePath = filePath.substring(0, filePath.lastIndexOf("/"));
+
+  if (tdarrPath || plexPath) {
+    filePath = filePath.replace(tdarrPath, plexPath)
+  }
+
+  response.infoLog += `Attempting to update Plex path ${filePath} in library ${key}\n`
+
+  const fullUrl = `${type}://${url}${port ? `:${port}` : ''}/library/sections/${key}/refresh?path=${filePath}&X-Plex-Token=${token}`
+  const redactedUrl = `${type}://${url}${port ? `:${port}` : ''}/library/sections/${key}/refresh?path=${filePath}&X-Plex-Token=[redacted]`
+
+  if (type === 'http') {
+    http.get(fullUrl, (res) => {
+      checkReply(res.statusCode, redactedUrl)
+      return response
+    }).on('error', (e) => {
+      response.infoLog += `We have encountered an error: ${e}`
+      return response
+    });
+  } else if (type === 'https') {
+    https.get(fullUrl, (res) => {
+      checkReply(res.statusCode, redactedUrl)
+      return response
+    }).on('error', (e) => {
+      response.infoLog += `We have encountered an error: ${e}`
+      return response
+    });
+  } else {
+    response.infoLog += `Plex could not be updated, the Url_Protocol can only be http or https. ${type} is not valid \n`
+    return response
+  }
+};
+
+function checkReply(statusCode, redactedUrl) {
+  if (statusCode == 200) {
+    response.infoLog += `☒ Above shown folder scanned in Plex! \n`
+  } else if (statusCode == 401) {
+    response.infoLog += `Plex replied that the token was not authorized on this server \n`
+  } else if (statusCode == 404) {
+    response.infoLog += `404 Plex not found, http/https is set properly? The URL used was 
+    ${redactedUrl} \n`
+  } else {
+    response.infoLog += `There was an issue reaching Plex. The URL used was 
+    ${redactedUrl} \n`
+  }
```

Comment:
Strict type checking using `===` instead of `==`

Reply:
i forgot to put it back in testing to figure out why there were never getting called. Though it may be doing something crazy with the responses. Then i figured out the call backs weren't working at all and never revisited.